### PR TITLE
feat.add image keywords as repo label.

### PR DESCRIPTION
**Pull request by lingdie**
feat.add image keywords as repo label.

### CODE REVIEW

File: controllers/imagehub/api/v1/repository_types.go [hunk 1]
```diff
@@ -103,6 +104,16 @@ func (r *Repository) getName() string {
 	return r.Name
 }
 
+func (r *Repository) genKeywordsLabels(img *Image) {
+	if r.Labels == nil {
+		r.Labels = make(map[string]string)
+	}
+	for _, keyword := range img.Spec.DetailInfo.Keywords {
```
Comment (zzjin) [inline controllers/imagehub/api/v1/repository_types.go:111]: the name is only supported by: 
```
The name segment is required and must be 63 characters or less, beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between. 
```
Direct use of `DetailInfo.Keywords` will fail duo to some wrong words?

Comment (lingdie) [reply controllers/imagehub/api/v1/repository_types.go:111]: solved. use a map to check.

File: controllers/imagehub/api/v1/common_types.go [hunk 2]
```diff
@@ -28,6 +31,19 @@ const (
 	defaultImagehubNamespace = "imagehub-system"
 )
 
+func GetSupportedKeywords() []string {
+	return []string{"Kubernetes", "Storage", "Network", "Database", "Monitoring", "Logging", "Dashboard", "MQ", "Platform", "GPU", "GitOps"}
+}
+
+var supportedKeywordsMap map[string]struct{}
```
Comment (zzjin) [inline controllers/imagehub/api/v1/common_types.go:38]:
```go
var supportedKeywordsMap = func() map[string]struct{} {
return xxxx;
}()
```
Make map var simper at build time not init runtime.

Comment (lingdie) [reply controllers/imagehub/api/v1/common_types.go]: done.

File: controllers/imagehub/api/v1/repository_types.go [hunk 3]
```diff
@@ -103,6 +104,19 @@ func (r *Repository) getName() string {
...
+		if _, ok := supportedKeywordsMap()[keyword]; ok {
```
Comment (zzjin) [inline controllers/imagehub/api/v1/repository_types.go:112]: Do not call func every time under for loop.

**Review by zzjin — APPROVED** [review]

**Pull request merged**